projects/ADuCM3029_demo_cn0428_cn0429/src/cli.c

```diff
@@ -0,0 +1,986 @@
+		p++;
+	}
+
+	return p;
+}
+
+/**
+   @brief Separates a command line argument
+
+    @param dst - pointer to a buffer where the argument will be copied
+    @param args - pointer to the current position of the command line .
+
+   @return none
+
+**/
+void GetArgv(char *dst, uint8_t *args)
+{
+	uint8_t  *s = args;
+	char     *d = dst;
+
+	while (*s) {
+		if (*s == _SPC)
+			break;
+		*d++ = *s++;
+	}
+
+	*d = '\0';
+}
+
+/**
+   @brief Puts command for FSM to the ring buffer
+
+   @param cmd - pointer to the command
+
+   @return none
+**/
+void CmdToRB(const char *cmd)
+{
+	uint8_t cmdLen = strlen(cmd);
+	for (int i = 0; i < cmdLen; i++) {
+		*RingBuf_Putvalue(&RX) = *cmd++;
+		RingBuf_Write_Increment(&RX);
+	}
+	Empty_Ring();
+}
+
+/**
+   @brief Command line process function
+
+   @return none
+**/
+void CmdProcess(void)
+{
+	cmdFunc   func;
+
+	/* <ENTER> key was pressed */
+	/* Find needed function based on typed command */
+	func = FindCommand((char *)cmdInString);
+
+	/* Check if there is a valid command */
+	if (func) {
+		/* Call the desired function */
+		(*func)(&cmdInString[2]);
+
+		/* Check if there is no match for typed command */
+	} else if (strlen((const char *)cmdInString) != 0) {
+		/* Display a message for unknown command */
+		UART_TX("Unknown command!"_EOS);
+	}
+	/* Prepare for next <ENTER> */
+	// uart_cmd = UART_FALSE;
+	CmdPrompt();
+}
+
+/**
+   @brief Command line prompt. Caller must verify that only the enabled board can run this.
+
+   @return int - UART status: UART_SUCCESS or error status
+**/
+void CmdPrompt(void)
+{
+//	int res;
+	static uint8_t count = 0;
+
+
+	/* Check first <ENTER> is pressed after reset */
+	if(count == 0) {
+		UART_TX("Welcome to CN0429!"_EOS);
+		UART_TX("Type <help> to see available commands..."_EOS);
+		count++;
+	}
+
+}
+
+/**
+   @brief Find available commands
+
+   @param cmd - command to search
+
+   @return cmdFunc - return the specific function for available command or NULL
+					 for invalid command
+**/
+cmdFunc FindCommand(char *cmd)
+{
+	cmdFunc func = NULL;
+	char *p = cmd;
+	int i = 0;
+	int f1 = 0;
+	int cmdlength = 0;
+
+	while(*p != 0) {
+		if (*p == _SPC)
+			f1 = 1;
+		else if (!f1)
+			cmdlength++;
+		p++;
+	}
+
+	while (CmdFun[i] != NULL) {
+		if(strncmp(cmd, CmdCommands[i], cmdlength) == 0) {
+			func = CmdFun[i];
+			break;
+		}
+		i++;
+	}
+
+	return func;
+}
+
+void DoNothing()
+{
+	// nothing to do here
+	// this is to avoid help printing each time enter is pressed
+}
+
+/**
+   @brief Set sensor to use for single sensor commands
+
+   @param args - pointer to the arguments on the command line.
+   	   site: 1, 2, 3, or 4
+
+   @return none
+**/
+void SetDefaultSns(uint8_t *args)
+{
+	uint8_t  *p = args;
+	char     arg[17];
+	char	 buff[20] = {0};
+
+	/* Check if this function gets an argument */
+	while (*(p = FindArgv(p)) != '\0') {
+		/* Save channel parameter */
+		GetArgv(arg, p);
+	}
+
+	/*DEFAULT CONFIGURATION*/
+	if (strncmp(arg, "1", 2) == 0) {
```

Comment:
there are C library functions to parse integers;
so, in this case you could use either `atoi` or `stroul` ;
so, something like 
```
int tmp = atoi(arg);
if (tmp >= 1 && tmp  <= 4) {
    uiDefaultSite = tmp;
    uiDefaultAddress = ADI_CFG_I2C_SENSOR1_ADDR + tmp;
}

Reply:
I don't think we can always guarantee that the addresses will be sequential. Users can change the address configuration if there is conflict with other i2c slaves on the bus. But I was putting the addresses in an array based on a previous comment, so we should be able to use that. 

I thought I had heard that atoi was bad practice due to undefined behavior and no error checking. Is that true?